src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -182,6 +182,38 @@ object Scanners {
      */
     var revComments: List[Comment] = Nil
 
+    /** The currently closest docstring, replaced every time a new docstring is
+     *  encountered
+     */
+    var closestDocString: List[List[Comment]] = List(List())
```

Comment:
So let's say we have this scenario:

```scala
/** Docstring 1*/
object O {
  /** Docstring 2 */
  class Inner

  /** Cheeky docstring */
}

class Outer
```

The class `Outer` should not have a docstring. So when entering / exiting blocks - I push and pop the docstring environment.

But sure - a better name for it would be something like  ~ `docStringsForBlock`